components/transactions/TransactionItem.js

```diff
@@ -305,18 +307,43 @@ const TransactionItem = ({ displayActions, collective, transaction, onMutationSu
             </StyledButton>
           </Container>
         )}
-        {hasExpense && (
-          <Container mt={3} pt={[2, 0]}>
+        {isExpense && (
+          <Container display="flex" mt={3} pt={[2, 0]}>
             <ExpenseTags expense={expense} />
+            <StyledButton
+              data-cy="expense-details"
+              buttonSize="tiny"
+              buttonStyle="secondary"
+              isBorderless
+              onClick={() => setExpanded(!isExpanded)}
+            >
+              <Span whiteSpace="nowrap">
+                {isExpanded ? (
+                  <React.Fragment>
+                    <FormattedMessage id="closeDetails" defaultMessage="Close Details" />
+                    &nbsp;
+                    <ChevronUp size="1em" />
+                  </React.Fragment>
+                ) : (
+                  <React.Fragment>
+                    <Span whiteSpace="nowrap">
+                      <FormattedMessage id="viewDetails" defaultMessage="View Details" />
+                      &nbsp;
+                      <ChevronDown size="1em" />
+                    </Span>
+                  </React.Fragment>
+                )}
+              </Span>
+            </StyledButton>
```

Comment:
Let's not duplicate this code please.

Reply:
Extracted it to a common function `transactionDetailLink` and avoided the duplication.